### PR TITLE
[LLVMGPU][ROCm] Disable kernarg preloading on pre-CDNA3 targets

### PR DESCRIPTION
**Pull request by kuhar**
This is not supported on CDNA1/CDNA2 devices which, depending on the firmware, may fail at runtime when executing code that uses kernel argument preloading.

I considered adding this to the `TargetWgp` attribute but decided against it because of how rocm-specific this optimization is.

Tested manually for a bunch of targets and looking at the ISA, including: `gfx90a`, `gfx940`, `gfx942`, and `gfx1100`.

### CODE REVIEW

File: compiler/plugins/target/ROCM/ROCMTarget.cpp [hunk 1]
```diff
@@ -132,6 +133,24 @@ struct ROCmOptions {
...
+  IREE::GPU::TargetAttr gpuTarget = getGPUTargetAttr(targetAttr);
+  if (!gpuTarget)
+    return failure();
+
+  StringRef arch = gpuTarget.getArch();
+  if (!arch.consume_front("gfx"))
+    return failure();
+
+  unsigned gfxVersion = 0;
+  if (!llvm::to_integer(arch, gfxVersion, 16))
```
Comment (krzysz00) [inline compiler/plugins/target/ROCM/ROCMTarget.cpp]: 1. I'm pretty sure this function is hanging out elsewhere in the code
2. The parsing isn't usually base-16. It's the rather annoying "last two digits are in hex, everything before that is decimal"

Comment (krzysz00) [reply compiler/plugins/target/ROCM/ROCMTarget.cpp]: (and we might want upstream's `amdgpu::Chipset` if we're feeling weirdly principled)

Comment (kuhar) [reply compiler/plugins/target/ROCM/ROCMTarget.cpp]: Oh you mean this helper? https://github.com/llvm/llvm-project/blob/e04d124a96311d3714522125bf703950863e0540/mlir/include/mlir/Dialect/AMDGPU/Utils/Chipset.h

Thanks, I didn't know we have it already.

**Review by krzysz00 — APPROVED** [review]
LGTM, thanks for chasing down all this

**Review by Groverkss — APPROVED** [review]
LGTM

**Review by antiagainst — APPROVED** [review]

**Pull request merged**